alacritty_terminal/src/tty/mod.rs

```diff
@@ -81,3 +79,57 @@ pub fn setup_env<C>(config: &Config<C>) {
         env::set_var(key, value);
     }
 }
+
+fn is_alacritty_terminfo_available() -> bool {
+    use std::fs;
+    use std::path::{Path, PathBuf};
+
+    let name = "alacritty";
+    let first = 'a';
+    let mut search: Vec<PathBuf> = Vec::new();
+    if let Some(dir) = env::var_os("TERMINFO") {
+        search.push(dir.into());
+    } else if let Some(mut home) = dirs::home_dir() {
+        home.push(".terminfo");
+        search.push(home);
+    }
+    if let Ok(dirs) = env::var("TERMINFO_DIRS") {
+        for dir in dirs.split(':') {
+            search.push(dir.into());
+        }
+    }
+    if let Ok(prefix) = env::var("PREFIX") {
+        let path = Path::new(&prefix);
+        search.push(path.join("etc/terminfo"));
+        search.push(path.join("lib/terminfo"));
+        search.push(path.join("share/terminfo"));
+    }
+    search.push("/etc/terminfo".into());
+    search.push("/lib/terminfo".into());
+    search.push("/usr/share/terminfo".into());
+    search.push("/boot/system/data/terminfo".into());
+    for path in search {
+        if fs::metadata(&path).is_ok() {
+            {
+                let mut path = path.clone();
+                path.push(first.to_string());
+                path.push(name);
```

Comment:
It might be possible to just change the way these paths are pushed to push the correct paths immediately using maybe a function as abstraction? That way we only need to do `paths.any(PathBuf::exists)` basically.